### PR TITLE
fix stylecheck bug with nre

### PR DESCRIPTION
**Pull request by ringabout**
Before `nim r --stylecheck:error --eval:"import nre"`

```
nre.nim(144, 22) Error: 'pcre' should be: 'Pcre' [type declared in C:\Users\blue\.choosenim\toolchains\nim-#devel\lib\wrappers\pcre.nim(257, 3)]   
```

```nim
type
  Regex* = ref object
    pattern*: string
    pcreObj: ptr pcre.Pcre  ## not nil
    pcreExtra: ptr pcre.ExtraData  ## nil

    captureNameToId: Table[string, int]
```

### CODE REVIEW

File: compiler/linter.nim [hunk 1]
```diff
@@ -109,7 +109,8 @@ proc differs(conf: ConfigRef; info: TLineInfo; newName: string): string =
   var first = min(info.col.int, line.len)
   if first < 0: return
   #inc first, skipIgnoreCase(line, "proc ", first)
-  while first > 0 and line[first-1] in Letters: dec first
```
Comment (ringabout) [inline compiler/linter.nim]: The while statement intends to handle the stropping identifier, right?

Comment (Araq) [reply compiler/linter.nim]: Yes and iirc also operators.

Comment (ringabout) [reply compiler/linter.nim]: I see

File: compiler/linter.nim [hunk 2]
```diff
@@ -109,7 +109,8 @@ proc differs(conf: ConfigRef; info: TLineInfo; newName: string): string =
...
+  if first < line.len and line[first] != '.':
```
Comment (Araq) [inline compiler/linter.nim]: Why would it be a '.'?

File: compiler/linter.nim [hunk 3]
```diff
@@ -109,7 +109,8 @@ proc differs(conf: ConfigRef; info: TLineInfo; newName: string): string =
...
+    while first > 0 and line[first-1] in Letters: dec first
```
Comment (ringabout) [inline compiler/linter.nim]:
```suggestion
  if first < line.len:
    if line[first] != '.':
      while first > 0 and line[first-1] in Letters: dec first
    else:
      inc first
```

File: compiler/linter.nim (hunk 2 above)
Comment (ringabout) [reply compiler/linter.nim]: For `pcre.Pcre`, it is parsed into a `dot expr`. The col stops at the position of `.`, which is the line info of `Pcre` token. So the `first` variable need to increase one lest the `.` is accounted.

```nim
proc builtinFieldAccess(c: PContext, n: PNode, flags: TExprFlags): PNode =
  var s = qualifiedLookUp(c, n, {checkAmbiguity, checkUndeclared, checkModule})
  if s != nil:
    markUsed(c, n[1].info, s)
```

File: compiler/linter.nim (hunk 3 above)
Comment (Araq) [reply compiler/linter.nim]: The idea is to find the boundaries of an identifier, there is nothing special about '.', so this logic makes no sense.

**Comment by ringabout** [comment]
```nim
proc builtinFieldAccess(c: PContext, n: PNode, flags: TExprFlags): PNode =
    markUsed(c, n[1].info, s)   # <----------  dotexpr is already checked here
    result = semSym(c, n, s, flags) # we shouldn't let semSym check dotexpr again 
    onUse(n[1].info, s)
    return

**Comment by ringabout** [comment]
find another same bug

*serve.nim*
```nim
const Serve* = 12
```
*test.nim*

```nim
import serve

echo serve.Serve
```

`test.nim(9, 11) Error: 'serve' should be: 'Serve' [const declared in C:\Users\blue\Downloads\Nim\serve.nim(1, 7)]`

**Comment by ringabout** [comment]
Ready for review

**Comment by arnetheduck** [comment]
Can we get a backport? been hitting this too.

**Comment by ringabout** [comment]
> Can we get a backport? been hitting this too.

If you would like a commit to be backported (and it should), you can @narimiran in case that the message is ignored. Nowadays it seems that contributing to 1-6 branch directly is acceptable => https://github.com/nim-lang/Nim/pull/19697

**Comment by narimiran** [comment]
> Nowadays it seems that contributing to 1-6 branch directly is acceptable

If somebody (accidentally) does it, it doesn't mean it is acceptable/preferred. I would much rather have PRs targeted at `devel` and marked as `[backport]`.

**Comment by ringabout** [comment]
Though it happened multiple times:

https://github.com/nim-lang/Nim/pull/19583
https://github.com/nim-lang/Nim/pull/19269
https://github.com/nim-lang/Nim/pull/19047

Even the rule has already been written into the contributuing guide.

**Pull request merged**